### PR TITLE
feat(observe): scoped PR observation and review inbox

### LINKED ISSUE #104: feat(workflow): add scoped PR observation and review inbox ergonomics
### Context

**Part of #59** (Phase 2 — reinguard-native workflow FSM).

There are two related UX failures for AI agents working on PR review flows.

**Problem A — observation target drift:** `rgd context build` resolves the GitHub PR facet from the **current git branch**. When an agent is asked to work on a specific PR (e.g. by number) while checked out elsewhere, operational context (state, routes, guards, filtered knowledge) can describe the **wrong** PR. Today the JSON does not make the **observed PR / branch / head SHA** prominent enough for agents to catch user-intent mismatches early.

**Problem B — fragile review comment and thread UX:** `review-address` and related knowledge still assume agents **hand-assemble** `gh api` / GraphQL for:

- enumerating `reviewThreads` and `isResolved`
- posting threaded disposition replies (REST shape: `in_reply_to`, full `commit_id`, `path`, `line`)
- resolving threads (`resolveReviewThread`)

That duplicates normative shapes already documented in policy, is error-prone (pagination, wrong PR, short SHA 422s), and forces a **second hop** after `rgd observe` because observation today exposes **aggregates** (e.g. unresolved count) without a **stable per-thread read model** aligned with those API steps. The reviews facet already incurs GraphQL cost for thread pagination but does not surface thread ids and anchors agents need for the happy path.

**Product intent (single vertical slice):** improve agent UX by stabilizing **scoped observation**, **observed-scope visibility**, a **review inbox read model** (thread records with stable identifiers), and **minimal non-observe `rgd` commands** for reply/resolve transport — without moving review judgment or workflow orchestration into the substrate (ADR-0001).

### Non-goals

- No review judgment or disposition selection in `rgd`.
- No consensus automation, auto-resolve, or merge decisions beyond existing policy and guards.
- No replacement of Adapter procedures with orchestration inside `rgd`.
- No mutating behavior under `rgd observe` or the observe phase of `rgd context build`.
- No scope merge with **#102** (stale bot review, HS-NO-DISMISS, and related policy hardening stay separate).
- No requirement in this issue to model **non-thread** findings (outside-diff, PR summary, conversation-only) in the observation inbox; existing policy paths for those remain unless a follow-up issue extends the read model.

### Refs: ADR

- **ADR-0001** — Adapter / Semantics / Substrate; `rgd` stabilizes information space, not agent reasoning.
- **ADR-0003** — Pull-based, stateless CLI; explicit scope is per-invocation input, not durable session state inside `rgd`.
- **ADR-0006** — GitHub auth via `gh` only (Phase 1).
- **ADR-0009** — Observation engine: **`rgd observe` remains read-only**; execution helpers must be a **separate explicit command surface** if implemented in `rgd`.
- **ADR-0011** — Semantic control plane; **`docs/cli.md` is the CLI I/O SSOT** for flags and stdout fields; procedures live under `.reinguard/procedure/`.
- **ADR-0013** — FSM states/routes; scoped observation must not **silently skew** signal meaning relative to control rules when the observed PR differs from the checked-out branch.

### ADR Impact

- **Amend `docs/adr/0009-observation-engine-abstraction.md`** — Document that `observe` stays side-effect-free and that **review reply / resolve** (if shipped as `rgd` commands) live outside `observe` in an explicit namespace; observation semantics unchanged.
- **ADR-0013:** **none by default.** Amend only if implementation adds or changes `state_id` / `route_id` or documents new FSM invariants for scoped observation.
- **ADR-0001, ADR-0003, ADR-0011:** **none** if the boundary above is preserved.

### Acceptance ↔ ADR

- **ADR-0001:** Deliverables are structured observation + thin execution transport only; no workflow brain.
- **ADR-0003:** PR/branch scope is explicit per invocation; no daemon or cross-run state in `rgd`.
- **ADR-0009:** Read path stays read-only; mutations only via dedicated non-observe commands (names TBD in the implementing PR, documented in `docs/cli.md`).
- **ADR-0011:** All new CLI surface and JSON fields documented in `docs/cli.md`; `review-address` updated to prefer the new read model and commands for covered paths; consensus rules remain in policy, not encoded as heuristics in Go.
- **ADR-0013:** When scoped PR ≠ current branch, operational context must **show** which PR/SHA was observed so FSM/guards are not misread as applying to the wrong target.

#### Layer 1 — Scoped observation

- `rgd context build` (and workflow-facing observation entry points) accept **explicit PR and/or branch scope**; precedence, defaults, and interaction with `--observation-file` are documented in `docs/cli.md`.
- stdout (or observation document) includes **machine-readable observed-scope metadata** so agents can assert intent: at minimum local git branch at collect time, resolved PR number for GitHub facets when applicable, and PR head SHA used for those facets.
- Adapter **Sense** guidance in `.cursor/commands/rgd-next.md`: when the user names a PR number, prefer scoped `rgd context build` (or equivalent documented invocation) over implicit branch-only resolution.

#### Layer 2 — Review inbox (read-only, observation)

- Observation exposes **unresolved review thread** records with stable identifiers and anchors needed for disposition and resolve flows per `.reinguard/policy/review--consensus-protocol.md` (e.g. GraphQL thread node id, root comment database id, anchor path/line/commit, author/body on root, `isResolved` / `isOutdated` as applicable).
- Aggregate signals used by FSM and `merge-readiness` (e.g. `review_threads_unresolved`, pagination flags) remain correct or are intentionally extended without breaking existing control rules.

#### Layer 3 — Review actions (execution, not `observe`)

- `rgd` exposes a **minimal non-observe** command surface (exact subcommand names TBD in the implementing PR) for: threaded disposition reply and `resolveReviewThread`, with **transport and input validation only** — no new consensus policy in Go beyond safe errors on bad input.
- GitHub access remains **`gh`-mediated** per ADR-0006.

#### Cross-cutting

- `docs/cli.md` updated for all new flags, commands, and JSON fields; embedded schemas / golden fixtures updated if observation or context output shape changes (ADR-0008).
- `.reinguard/procedure/review-address.md` updated so the happy path does not require hand-built `gh api` snippets for the covered operations; optional cross-link from `.reinguard/knowledge/review--github-thread-api.md` where it reduces duplication.
- `rgd config validate` passes after any config/schema changes.

### Test plan

- Repo baseline: `go test ./... -race -count=1`, `go vet ./...`, `golangci-lint run`.
- Markdown: `pre-commit run markdownlint-cli2 --all-files` when `.md` files change.
- Config: `rgd config validate` (or `go run ./cmd/rgd config validate` if `rgd` is not on `PATH`).
- **Behavioral intent:** explicit `--pr` (or equivalent) makes `github.pull_requests.*` and review signals match that PR’s ground truth vs default branch-based resolution; observed-scope metadata is testable in fixtures; inbox thread list matches GraphQL ground truth for a representative case; execution helpers succeed against httptest or documented manual checks without violating read-only `observe`.

### Linked issues

- Parent: #59
- Related (do not merge scopes): #102

### Notes

The goal is fewer **wrong-target** contexts and fewer **raw API assembly** mistakes for AI agents, while keeping semantic judgment and consensus obligations in Semantics + Adapter policy, not in substrate heuristics.

### PR DESCRIPTION
**Pull request by k-shibuki**
## Summary

- Add `--branch` / `--pr` scope for live observation and `rgd context build`, with `observed_scope` and `review_inbox` signals.
- Add `rgd review reply-thread` and `rgd review resolve-thread` as thin `gh api` transports (observation remains side-effect free).
- Document precedence when both `--pr` and `--branch` are set; tighten validation and inbox edge cases.

## Traceability

Closes #104

Refs: #104

## Definition of Done

- [x] Tests added or updated (`go test ./...`)
- [x] `go vet ./...` clean
- [x] Lint clean (golangci-lint / CI)
- [x] Documentation updated if behavior or public CLI surface changed

## Test plan

1. `go test -race ./...` — all packages pass.
2. `go vet ./...` and `golangci-lint run` — clean.
3. `bash .reinguard/scripts/with-repo-local-state.sh -- pre-commit run markdownlint-cli2 --all-files` — clean.
4. `go run ./cmd/rgd config validate` — OK.
5. Local CodeRabbit: `bash .reinguard/scripts/with-repo-local-state.sh --home-subdir cr-home -- bash .reinguard/scripts/check-local-review.sh --base main --retry-on-rate-limit` — completed with exit 0 on latest head; findings dispositioned (docs/tests/validation fixes applied; optional items documented or deferred as nitpicks).

## Risk / Impact

- **Observation / CLI**: New flags and signal shapes; callers using saved observation files are unchanged except documented scope rules.
- **Review transport**: New commands shell out to `gh` only; same auth contract as the rest of `rgd` (ADR-0006).

## Rollback Plan

- Revert this PR on `main` via GitHub revert or `git revert` of the merge commit; no migration of stored data required.

## Exception

- Type:
- Justification:

### CODE REVIEW

File: .reinguard/scripts/lib/json_minimal.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+json_escape() {
+  local value="$1"
+  value=${value//\\/\\\\}
+  value=${value//\"/\\\"}
+  value=${value//$'\n'/\\n}
+  value=${value//$'\r'/}
+  printf '%s' "$value"
+}
```
[the remark this answers is not in the document]
Comment (k-shibuki) [reply .reinguard/scripts/lib/json_minimal.sh]: **Fixed.** Escaped tab characters in `json_escape` for valid JSON when values contain literal tabs.

**Comment by k-shibuki** [comment]
@coderabbitai review
